### ISSUE TITLE
Alt-F2 "run command"-box sometimes loses keyboard focus to desktop

### ISSUE THREAD

**Issue by hlinden** [linuxmint/Cinnamon#5937]
I can not deterministically reproduce this - however, since my last update, the "run command" box that opens when you press Alt-F2 sometimes loses the keyboard, and on even rarer occasions also the mouse focus to the desktop. That is a problem, because I can not close it then, the "Menu" button does not react and I have the overlay all over my desktop. The only solution is to log in at a console and reboot the entire machine.

I am running Cinnamon 3.2.1 and  X.Org X Server 1.18.4 on Arch Linux.
What can I do to trace this problem properly?

**Comment by vladimir-didenko** [comment]
I have similar problem with Alt-Tab. Sometimes application list loses focus but still stays visible until I click on panel.

**Comment by hlinden** [comment]
Can confirm that as well. Happened once yesterday.

**Comment by erik-mansson** [comment]
Upgraded to Cinnamon 3.2.1 a few days ago. A moment ago I started getting problems with the interface. If it matters I can add that I saw these quite soon after having locked my screen and resumed working:

1. (This bug) Alt+F2 gives the run/launcher dialog box without focus. I can still use the mouse and keyboard for programs in the background, although they are grayed out a bit by an overlay.

2. When opening start menu (e.g. by windows-key) although focus was in the text field, trying to type the name of an application had no effect.

3. Alt+Tab shows the selector with a row of icons centered on the screen, and it doesn't go away after I've stepped to the window I want to switch to. By clicking around and re-selecting I somehow get it to disappear after several attempts each time, but it reoccurred every time I was using Alt+Tab.

To solve the Alt+F2 problem I opened a terminal, and using "ps aux" and some grepping I took a guess and killed cinnamon-screensaver-main.py (no effect). Then I killed python3 /usr/bin/cinnamon-launcher and this removed the task bar and all windows for a second (showing desktop background only) then Cinnamon restarted. All my windows were restored and the Alt+F2 overlay disappeared. In the attached log file this is "Cinnamon started at Sun Nov 20 2016 18:37:44".

This also solved all the other problems! 
1. Alt+F2 now shows a text cursor and allows me to type.
2. Now I can type into the text field to filter programs from the start menu. 
3. Alt+Tab's normal immediate switching works again, only showing the list of icons if I keep the keys pressed for a while (and it goes away when I release the keys).

[xsession-errors_2016-11-20.txt](https://github.com/linuxmint/Cinnamon/files/602199/xsession-errors_2016-11-20.txt)
